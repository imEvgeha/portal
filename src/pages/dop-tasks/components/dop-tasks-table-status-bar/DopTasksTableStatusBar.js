import React from 'react';
import PropTypes from 'prop-types';
import './DopTasksTableStatusBar.scss';

const DopTasksTableStatusBar = ({paginationData}) => {
    return (
        <div className="nexus-c-dop-tasks-table-status-bar">
            <span className="nexus-c-dop-tasks-table-status-bar__description">
                Rows:{' '}
                <span className="nexus-c-dop-tasks-table-status-bar__value">
                    {paginationData.pageSize} of {paginationData.totalCount}
                </span>
            </span>
        </div>
    );
};

DopTasksTableStatusBar.propTypes = {
    paginationData: PropTypes.object,
};

DopTasksTableStatusBar.defaultProps = {
    paginationData: {
        pageSize: 0,
        totalCount: 0,
    },
};

export default DopTasksTableStatusBar;
