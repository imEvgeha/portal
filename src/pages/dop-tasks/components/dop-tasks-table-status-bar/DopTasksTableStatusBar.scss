@import '../../../../styles/config.scss';
@import '../../../../styles/atlaskit/mixins';

.nexus-c-dop-tasks-table-status-bar {
    display: flex;
    align-items: center;
    width: 100%;
    background-color: map-get($colors, nexus-white);
    border-bottom: 1px solid map-get($colors, nexus-light-grey);
    border-left: 1px solid map-get($colors, nexus-light-grey);
    border-right: 1px solid map-get($colors, nexus-light-grey);
    @include addSpacing(height, 6);
    @include heading(h200);
    @include margin(0);
    @include padding(0, 0, 0, 2);
}
