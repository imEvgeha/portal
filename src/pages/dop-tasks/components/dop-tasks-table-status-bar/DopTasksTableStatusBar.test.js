import React from 'react';
import {shallow} from 'enzyme';
import DopTasksTableStatusBar from './DopTasksTableStatusBar';

describe('DopTasksTableStatusBar', () => {
    let wrapper = null;

    const props = {
        paginationData: {
            pageSize: 0,
            totalCount: 0,
        },
    };

    beforeEach(() => {
        wrapper = shallow(<DopTasksTableStatusBar {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render status bar message', () => {
        const statusDescription = wrapper.find('.nexus-c-dop-tasks-table-status-bar__description');
        expect(statusDescription.text()).toEqual('Rows: 0 of 0');
    });
});
