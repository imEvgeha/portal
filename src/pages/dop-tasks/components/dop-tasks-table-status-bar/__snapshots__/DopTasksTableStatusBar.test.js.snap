// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DopTasksTableStatusBar should match snapshot 1`] = `
<div
  className="nexus-c-dop-tasks-table-status-bar"
>
  <span
    className="nexus-c-dop-tasks-table-status-bar__description"
  >
    Rows:
     
    <span
      className="nexus-c-dop-tasks-table-status-bar__value"
    >
      0
       of 
      0
    </span>
  </span>
</div>
`;
